### ISSUE TITLE
Eliminate all python2 compatibility concerns from the codebase

### ISSUE THREAD

**Issue by d-w-moore** [irods/python-irodsclient#480]
Starting at v2.0.0, we plan to no longer adhere to any sort of standard for Python2 compatibility.  

More than than, we indent to expurgate the code base of any signs the Python iRODS Client ever ran on Python 2.x ....

At the least, this means:
   - deleting all python2-isms esp. if they complicate or obfuscate otherwise clear, understandable code.
   - not having to use the six module or recast code to account for the possibility of running on Python2
   - not duplicating tests for the purpose of testing Python2 compatibility
   - removing atrocious necessities such as `MyIntEnum`
   - undoubtedly even more such changes as benefit developer sanity

**Comment by trel** [comment]
Yes please.     Only as part of a new 'major' release of this client library.

**Comment by d-w-moore** [comment]
Note the guidelines in the description here are to be applied only in changes made after `v3.0.0` has been released (projected to be our first Python3-only release).   Format-strings (f"...") particularly should be disallowed til then, as the presence of even one of these in a source file prevents it from being parsed by the Python2 interpreter.
